controller/src/util/instance_action.rs

```diff
@@ -419,20 +419,27 @@ pub async fn handle_instance_change(
         }
     };
     if let Some(broker_spec) = &configuration.spec.broker_spec {
-        match broker_spec {
+        let instance_change_result = match broker_spec {
             BrokerSpec::BrokerPodSpec(p) => {
                 handle_instance_change_pod(instance, p, action, kube_interface).await
             }
             BrokerSpec::BrokerJobSpec(j) => {
                 handle_instance_change_job(
-                    instance.clone(),
+                    instance,
                     *configuration.metadata.generation.as_ref().unwrap(),
                     j,
                     action,
                     kube_interface,
                 )
                 .await
             }
+        };
+        match instance_change_result {
```

Comment:
i wonder if this can be simplified (forgive my terrible rust memory):

```rust
    if let Some(broker_spec) = &configuration.spec.broker_spec {
        match broker_spec {
        let instance_change_result = match broker_spec {
            . . .
        };
        if let Err(e) = match instance_change_result {
            error!("Unable to handle Broker action: {:?}", e);
        }
    } 
    Ok(())
}
```

Reply:
Indeed, way easier to read now, thanks for pointing it.

Reply:
lol ... might even be able to combine the `let instnace_change_result` block with the `if let Err(e)` block.  :)